returnn/tf/layers/rec.py

```diff
@@ -5010,7 +5017,11 @@ def get_dep_layers(self):
     # See also self.transform_config_dict where we might strip away the sources.
     ls = super(ChoiceLayer, self).get_dep_layers()
     if self.explicit_search_sources:
-      ls.extend(self.explicit_search_sources)
+      if isinstance(self.explicit_search_sources, dict):
+        additional_sources = self.explicit_search_sources.values()
```

Comment:
This is non-deterministic in earlier Python versions.
Better:
```suggestion
        additional_sources = [src for _, src in sorted(self.explicit_search_sources.items())]
```